pkg/service/tenant.go

```diff
@@ -42,15 +43,19 @@ func (s *APIImpl) CreateTenant(c echo.Context) error {
 		*newTenant.GitRepo.Url == "" {
 		return echo.NewHTTPError(http.StatusBadRequest, "GitRepo URL is required")
 	}
-	apiTenant := &api.Tenant{
-		TenantProperties: api.TenantProperties(*newTenant),
+	apiTenant := api.Tenant(*newTenant)
+	if !strings.HasPrefix(string(apiTenant.Id), api.TenantIDPrefix) {
```

Comment:
I'm seeing this `string(id)` more often now, which kind of annoys me a little. I think it makes sense to implement the `Stringer` interface for the id like
```go
func (in Id) String() string {
  return string(in)
}
```
```suggestion
	if !strings.HasPrefix(apiTenant.Id.String()), api.TenantIDPrefix) {
```
Shouldn't bee too difficult.
You could also do that in a separate PR if you would like to separate this change.

Reply:
Ah, got point. Totally missed that one.